jenkins/check_copyright.py

```diff
@@ -42,6 +42,8 @@ def report_incorrect(file_name, pairs):
 linecount = 0
 pairs = []
 with open(file_name, 'r') as sourcefile:
+    if not re.match(r'#!/usr/bin/env.*', sourcefile.readline(), 0):
+        sourcefile.seek(0)
```

Comment:
Actually I like your sourcefile.seek method. Can you change this to be more general as follows?

```python
pairs = []
with open(file_name, 'r') as sourcefile:
    shebang = sourcefile.readline()
    if not shebang.startswith("#!"):
        sourcefile.seek(0)
    pairs += zip(regexes, sourcefile)
```